### PR TITLE
[Rasheed]Building a chat dapp with nextjs web3js and bigquery

### PR DESCRIPTION
**Pull request by Taiwrash**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

**Review by yafiabiyyu — COMMENTED** [review]
‌In general, your tutorial is good. However, there are still some areas that need improvement. I've provided suggestions for improvement via your pull request.

File: blog/06-05-23-building-a-chat-dapp-with-nextjs-web3js-and-bigquery/building-a-chat-dapp-with-nextjs-web3js-and-bigquery.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+In this tutorial, we explored how to build a decentralized chat DApp using Next.js, web3.js, and BigQuery. We started by setting up a Celo network and deploying a smart contract for the chat functionality. Then, we used web3.js to connect to the Celo network and interact with the smart contract. Next, we integrated Pub/Sub to push chat events to BigQuery for analysis. Finally, we demonstrated how to query the chat data in BigQuery to find the most active users and the most messages sent. Here is a step-by-step guide on how to build a chat DApp using Next.js and Web3.js on the Celo network, and set up a BigQuery system to analyze the chat data:
+
+This tutorial serves as a guide for developers interested in building decentralized chat applications using the Celo network and leveraging the power of Google Cloud's BigQuery for data analysis. The provided sample codes can be adjusted and customized to meet specific use cases and requirements.
+
+## Prerequisite
+
+- GCP account
+- Celo wallet
+- Node.js
```
[the remark this answers is not in the document]
Comment (Taiwrash) [reply blog/06-05-23-building-a-chat-dapp-with-nextjs-web3js-and-bigquery/building-a-chat-dapp-with-nextjs-web3js-and-bigquery.md:26]: Nodejs is included already

File: blog/06-05-23-building-a-chat-dapp-with-nextjs-web3js-and-bigquery/building-a-chat-dapp-with-nextjs-web3js-and-bigquery.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      <input type="text" value={content} onChange={(e) => setContent(e.target.value)} />
+      <button onClick={handleSend}>Send</button>
+    </div>
+  );
+}
+export default Chat;
+```
+In this example, the Chat component uses the useState hook to manage the content state. When the user clicks the "Send" button, the handleSend function is called, which executes the sendMessage function on the smart contract and publishes a message to the Pub/Sub topic.
+
+### Creating an index JS
```
[the remark this answers is not in the document]
Comment (Taiwrash) [reply blog/06-05-23-building-a-chat-dapp-with-nextjs-web3js-and-bigquery/building-a-chat-dapp-with-nextjs-web3js-and-bigquery.md:140]: Yes it was intentional. It is a completion of the NextJS part of the article.

**Review by Ishanpathak1 — APPROVED** [review]
Approved @developerkunal

**Pull request merged**